### PR TITLE
Vet clinic database: query and update animals table

### PR DESCRIPTION
**Pull request by jlcarrascof**
This PR includes:

- [ ]     Add a column species of type string

- [ ]     Insert the following data:

Animal: His name is Charmander. He was born on Feb 8th, 2020, and currently weighs -11kg. He is not neutered and he has never tried to escape.
Animal: Her name is Plantmon. She was born on Nov 15th, 2021, and currently weighs -5.7kg. She is neutered and she has tried to escape 2 times.
Animal: His name is Squirtle. He was born on Apr 2nd, 1993, and currently weighs -12.13kg. He was not neutered and he has tried to escape 3 times.
Animal: His name is Angemon. He was born on Jun 12th, 2005, and currently weighs -45kg. He is neutered and he has tried to escape once.
Animal: His name is Boarmon. He was born on Jun 7th, 2005, and currently weighs 20.4kg. He is neutered and he has tried to escape 7 times.
Animal: Her name is Blossom. She was born on Oct 13th, 1998, and currently weighs 17kg. She is neutered and she has tried to escape 3 times.
Animal: His name is Ditto. He was born on May 14th, 2022, and currently weighs 22kg. He is neutered and he has tried to escape 4 times.

- [ ]     Add following transactions

1. Animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.

2. Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
Verify that changes were made.
Commit the transaction.
Verify that changes persist after commit.

3.Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)

4.Delete all animals born after Jan 1st, 2022.
Create a savepoint for the transaction.
Update all animals' weight to be their weight multiplied by -1.
Rollback to the savepoint
Update all animals' weights that are negative to be their weight multiplied by -1.
Commit transaction

5.Write queries to answer the following questions:
How many animals are there?
How many animals have never tried to escape?
What is the average weight of animals?
Who escapes the most, neutered or not neutered animals?
What is the minimum and maximum weight of each type of animal?
What is the average number of escape attempts per animal type of those born between 1990 and 2000?

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -6,4 +6,15 @@ VALUES
     ('Gabumon', '2018-11-15', 2, true, 8),
     ('Pikachu', '2021-01-07', 1, false, 15.04),
     ('Devimon', '2017-05-12', 5, true, 11);
+
+INSERT INTO animals (name, date_of_birth, escape_attempts, neutered, weight_kg, species)
```
Comment (t-yanick) [inline data.sql]: - [ ] Great job working on this milestone. The data you are to insert has the following fields `name, date_of_birth, escape_attempts, neutered, weight_kg`. Your inserted data has the `species` field which is not a requirement. Kindly update your fields.

File: queries.sql [hunk 2]
```diff
@@ -15,3 +15,51 @@ SELECT * FROM animals WHERE neutered = true;
...
+DELETE FROM animals;
+SELECT * FROM animals;
+ROLLBACK;
+SELECT * FROM animals;
+
+BEGIN;
+DELETE FROM animals WHERE date_of_birth > '2022-01-01';
+SAVEPOINT mysavepoint;
+UPDATE animals SET weight_kg = weight_kg * -1;
+ROLLBACK TO my_savepoint;
```
Comment (t-yanick) [inline queries.sql]: - [ ] The name of the SAVEPOINT you created is `mysavepoint` and in line 42, you do a ROLLBACK TO `my_savepoint.` I think there is some name discrepancy here. Kindly correct the name.

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @jlcarrascof , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  All SQL commands added to the correct files
:heavy_check_mark: Screenshots added to PR body
:heavy_check_mark:  Use of correct gitflow
:heavy_check_mark:  Good commit messages

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 2 above)
Comment (jlcarrascof) [reply queries.sql]: Yes, sir .... my bad, when I copied the original command to the script file. I fixed it.

FIX-02

And this is the prove that I ran my script well, when I was doing the exercise. This screenshot is above in the pull request's body.

269510735-ac2bb854-33fe-4b17-a3a5-59d6d54f9f9e

Thanks for your suggestions, @t-yanick dear reviewer.

File: data.sql (hunk 1 above)
Comment (jlcarrascof) [reply data.sql]: Ohh boy!! .... You're reason, sir .... I fixed the script in my **data.sql** file

FIX-01

**Review by topeogunleye — APPROVED** [review]
### Hi @jlcarrascof, 

# Status: Approved ✔️
Excellent job on making the changes requested by the previous code reviewer. Keep it up 👏

Your project is complete! There is nothing else to say other than... it's time to merge it 💪
Congratulations! 🎉
congratulations
>
## To highlight :champagne: 
### Code Review Highlights for Student's Updated Project

Hello 👋,

I've reviewed the revised version of your project, and I want to commend you for addressing the previous reviewer’s concerns diligently. Here are some highlights:

#### Highlights 🌟

1. **Data Integrity and Accuracy**: Great job updating the fields and aligning them correctly with the project requirements! This ensures that only the necessary data is included, maintaining the integrity and the cleanliness of the database. 👌

2. **Attention to Detail**: Correcting the discrepancy in the SAVEPOINT name demonstrates a high level of attention to detail. It’s crucial to maintain consistency in naming conventions, and your adjustment ensures that the SAVEPOINT and ROLLBACK operations interact as intended. 🛠️

3. **Quick Adaptation to Feedback**: It’s notable that you quickly and efficiently corrected the mentioned issues from the previous review. Swiftly adapting and applying feedback is an essential skill in software development, and you’ve demonstrated it well here. 🚀

4. **Proactive Error Resolution**: The corrections you made, particularly in the SAVEPOINT naming, will help in preventing potential errors and bugs that could have arisen due to the mismatch. Proactively addressing such issues is crucial in developing robust software. 🛡️

5. **Commitment to Excellence**: Your responsiveness to the review and your efforts to refine the details reflect your commitment to producing high-quality work, and this is very commendable! 🌟

Keep up the excellent work, and continue paying attention to the details; it truly makes a difference in the quality of your projects! Looking forward to seeing more of your progress.

## [Optional] suggestions:
* Nothing to mention.
 
 _Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
 
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @topeogunleye in your question so I can receive the notification.



------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by jlcarrascof** [comment]
Yes, sir **@topeogunleye** 

SAVEPOINT is ULTRA VERY IMPORTANT .... if you don't remind the name of your savepoint id, it'll be difficult to recover the info.

Thanks for your feedback, sir.

Javier.-
Venezuela.

**Comment by topeogunleye** [comment]
> Yes, sir **@topeogunleye**
> 
> SAVEPOINT is ULTRA VERY IMPORTANT .... if you don't remind the name of your savepoint id, it'll be difficult to recover the info.
> 
> Thanks for your feedback, sir.
> 
> Javier.- Venezuela.

Hello Javier,

Absolutely! Managing SAVEPOINT correctly is crucial for maintaining data integrity during database transactions. It's pivotal to remember or accurately reference the SAVEPOINT names to utilize them effectively. 🛠️

I'm glad the feedback was beneficial to you, and it's encouraging to see your attention to such vital details. Keep up the great work and continue honing your skills. 🚀

Best Regards,
Tope Ogunleye

**Pull request merged**